app/components/Editor.js

```diff
@@ -113,13 +117,40 @@ export default class Editor extends React.Component {
         this.redactor.insertion.insertHtml(replacement, 'start');
         this.redactor.selection.restoreMarkers();
       },
-      onReceiveNote: (_note) => {
-        // Empty
-      },
       clearUndoHistory: () => {
         // Called when switching notes to prevent history mixup.
         $R('#editor', 'module.buffer.clear');
       },
+      noteShouldBeRendered: async (note) => {
+        const renderNote = async () => {
+          const isUnsafeContent = this.checkIfUnsafeContent(note.content.text);
+          if (isUnsafeContent) {
+            const trustUnsafeContent = note.clientData['trustUnsafeContent'] ?? false;
+            if (!trustUnsafeContent) {
+              const result = await this.showUnsafeContentAlert();
+              if (result) {
+                this.setTrustUnsafeContent(note);
+              }
+              return result;
+            }
+          }
+          return true;
+        };
+
+        const enableEditing = await renderNote();
+        if (!enableEditing) {
+          $R('#editor', 'source.setCode', '');
+          $R('#editor', 'module.editor.disable');
+        } else {
+          $R('#editor', 'module.editor.enable');
+        }
+
+        this.setState({
```

Comment:
Having side effects in getters is a very dangerous idea.

Can this be refactored so that noteShouldBeRendered doesn't have consequences and doesn't change state?

Reply:
I've enabled read-only mode instead, to prevent making changes to the note's content.